### ISSUE TITLE
Constant 2 second latency on all packs requests

### ISSUE THREAD

**Issue by spazer5** [rails/webpacker#2566]
On a fresh Rails 6 or 5 install with `--webpack=vue` and on these specs with no gems/packages added:
- ruby 2.6.5
- windows 10 64bit

I'm experiencing a strange constant latency on every asset that's loaded through the webpack-dev-server proxy (meaning all `/packs` requests) have a constant 2 second latency.

And i'm confident this doesn't have to do with compiling the javascript, because even on subsequent refreshes with no code change, the latency's always there, but everything else like sprockets assets are served in 20ms or less.

![](https://s3.amazonaws.com/snaps-bucket/m/20/05/DknQUJHP/JTM9.png)

As you can see, no matter how often i refresh, everything is served instantly except those resources passing through the webpack proxy, they always serve at a strangely consistent 2 second latency.

I even tried to test a request directly to http://localhost:3035/ (webpack dev server) and it's actually much faster, so probably not something to do with /bin/webpack_dev_server, and more to do with the proxy on Rails.

Anyone having this same issue?

P.S. i run `ruby bin/webpack-dev-server` on top of `rails s`

**Comment by spazer5** [comment]
Temporarely resolved by running a node reverse proxy on top of Rails which routes `/packs` requests directly to webpacker dev server at `localhost:3035`.

This has dramatically eliminated the lag caused by the rack proxy.

Here's my `proxy.js` using the `http-proxy` package.

```
const railsPort   = 3000
const webpackPort = 3035

const http      = require('http')
const httpProxy = require('http-proxy')
const proxy     = httpProxy.createProxyServer()

http.createServer(function (req, res) {
  const port = req.url.match(/^\/packs\//i) ? webpackPort : railsPort

  proxy.web(req, res, {target: `http://localhost:${port}`})
}).listen(80)

proxy.on('proxyRes', function (proxyRes, req, res) {
  console.log('<<', `[${res.statusCode}]`, req.url)
})

console.log('[Rails <> Webpack] Node proxy running!')
```

**Comment by jakeNiemiec** [comment]
@spazer5 👍 I do something similar, except it is through the `browser-sync-webpack-plugin`. I agree the rails proxy _should_ be faster, although I (personally) don't know how you would speed it up.

**Comment by spazer5** [comment]
Good to know @jakeNiemiec I hope at some point this gets resolved and we could use the rack server as it's probably a more elegant solution than doing it this way.

**Comment by majkelcc** [comment]
I don't think this latency is normal - I just checked on my app (Rails 6, Webpacker 5) and `Webpacker::DevServerProxy` adds less than 10ms overhead (I'm running webpacker in a separate docker container in devevelopment).

@spazer5 about your proxy, couldn't you just change the `Rails.application.config.action_controller.asset_host` (in `config/environments/development`)  to `localhost:3035`? This way your application would produce urls hitting your webpacker dev server directly without any need for a proxy.
